<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>752</width>
    <height>547</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Serial port repeater</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QGroupBox" name="grpIn">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Input serial port</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="lblInPort">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cbInPort"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblInBaud">
             <property name="text">
              <string>Baud rate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="cbInBaud"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblInDataBits">
             <property name="text">
              <string>Data bits</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="cbInDataBits"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblInParity">
             <property name="text">
              <string>Parity</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="cbInParity"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lblInFlow">
             <property name="text">
              <string>Flow control</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="cbInFlow"/>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="lblInStopBits">
             <property name="text">
              <string>Stop bits</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QComboBox" name="cbInStopBits"/>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="lblInBytesReadedLabel">
             <property name="text">
              <string>Bytes read</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="lblInBytesReaded">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="lblInSpeedLabel">
             <property name="text">
              <string>Read speed</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="lblInSpeed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="lblInBytesWrittenLabel">
             <property name="text">
              <string>Bytes written</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QLabel" name="lblInBytesWritten">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="lblInWriteSpeedLabel">
             <property name="text">
              <string>Write speed</string>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLabel" name="lblInWriteSpeed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="12" column="0" colspan="2">
            <widget class="QCheckBox" name="cbxIn2Out1">
             <property name="text">
              <string>Input -&gt; Output 1</string>
             </property>
            </widget>
           </item>
           <item row="13" column="0" colspan="2">
            <widget class="QCheckBox" name="cbxIn2Out2">
             <property name="text">
              <string>Input -&gt; Output 2</string>
             </property>
            </widget>
           </item>
           <item row="14" column="0" colspan="2">
            <widget class="QCheckBox" name="cbxOut1ToIn">
             <property name="text">
              <string>Output 1 -&gt; Input</string>
             </property>
            </widget>
           </item>
           <item row="15" column="0" colspan="2">
            <widget class="QCheckBox" name="cbxOut2ToIn">
             <property name="text">
              <string>Output 2 -&gt; Input</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QLabel" name="lblInWriteErrors">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="lblInWriteErrorsLabel">
             <property name="text">
              <string>Write errors</string>
             </property>
            </widget>
           </item>
           <item row="11" column="0" colspan="2">
            <widget class="QCheckBox" name="cbDumpIn">
             <property name="text">
              <string>Dump to file</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>231</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="grpOut1">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Output 1 serial port</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="lblOut1Port">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cbOut1Port"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblOut1Baud">
             <property name="text">
              <string>Baud rate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="cbOut1Baud"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblOut1DataBits">
             <property name="text">
              <string>Data bits</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="cbOut1DataBits"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblOut1Parity">
             <property name="text">
              <string>Parity</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="cbOut1Parity"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lblOut1Flow">
             <property name="text">
              <string>Flow control</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="cbOut1Flow"/>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="lblOut1StopBits">
             <property name="text">
              <string>Stop bits</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QComboBox" name="cbOut1StopBits"/>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="lblOut1BytesReadedLabel">
             <property name="text">
              <string>Bytes read</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="lblOut1BytesReaded">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="lblOut1SpeedLabel">
             <property name="text">
              <string>Speed</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="lblOut1Speed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QLabel" name="lblOut1BytesWritten">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="lblOut1BytesWrittenLabel">
             <property name="text">
              <string>Bytes written</string>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLabel" name="lblOut1WriteSpeed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="lblOut1WriteSpeedLabel">
             <property name="text">
              <string>Write speed</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QLabel" name="lblOut1WriteErrors">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="lblOut1WriteErrorsLabel">
             <property name="text">
              <string>Write errors</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_2">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>231</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="grpOut2">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Output 2 seral port</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="lblOut2Port">
             <property name="text">
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="cbOut2Port"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="lblOut2Baud">
             <property name="text">
              <string>Baud rate</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QComboBox" name="cbOut2Baud"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="lblOut2DataBits">
             <property name="text">
              <string>Data bits</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QComboBox" name="cbOut2DataBits"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="lblOut2Parity">
             <property name="text">
              <string>Parity</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QComboBox" name="cbOut2Parity"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="lblOut2Flow">
             <property name="text">
              <string>Flow control</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QComboBox" name="cbOut2Flow"/>
           </item>
           <item row="6" column="1">
            <widget class="QLabel" name="lblOut2BytesReaded">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="lblOut2BytesReadedLabel">
             <property name="text">
              <string>Bytes read</string>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="lblOut2SpeedLabel">
             <property name="text">
              <string>Speed</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QLabel" name="lblOut2Speed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QComboBox" name="cbOut2StopBits"/>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="lblOut2StopBits">
             <property name="text">
              <string>Stop bits</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QLabel" name="lblOut2BytesWritten">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="lblOut2BytesWrittenLabel">
             <property name="text">
              <string>Bytes written</string>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QLabel" name="lblOut2WriteSpeed">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="9" column="0">
            <widget class="QLabel" name="lblOut2WriteSpeedLabel">
             <property name="text">
              <string>Write speed</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QLabel" name="lblOut2WriteErrors">
             <property name="frameShape">
              <enum>QFrame::Panel</enum>
             </property>
             <property name="frameShadow">
              <enum>QFrame::Sunken</enum>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="alignment">
              <set>Qt::AlignCenter</set>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="lblOut2WriteErrorsLabel">
             <property name="text">
              <string>Write errors</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <spacer name="verticalSpacer_3">
           <property name="orientation">
            <enum>Qt::Vertical</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>20</width>
             <height>231</height>
            </size>
           </property>
          </spacer>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pbStart">
        <property name="text">
         <string>Start</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pbStop">
        <property name="text">
         <string>Stop</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>752</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
